pilot/pkg/model/gateway.go

```diff
@@ -275,8 +275,10 @@ func resolvePorts(number uint32, instances []*ServiceInstance, legacyGatewaySele
 	}
 	ports := map[uint32]struct{}{}
 	for _, w := range instances {
-		if _, directPortTranslation := w.Service.Attributes.Labels[DisableGatewayPortTranslationLabel]; directPortTranslation {
+		if _, directPortTranslation := w.Service.Attributes.Labels[DisableGatewayPortTranslationLabel]; directPortTranslation && legacyGatewaySelector {
```

Comment:
is `legacyGatewaySelector` ever true? It seems like `gatewayWithInstances.legacyGatewaySelector` has no assignments

Reply:
I think its just your IDE missing the un-named assignment , I had the same problem.. https://github.com/istio/istio/blob/00f1548764e77b94276f49adc9539a6fad02c5ab/pilot/pkg/model/push_context.go#L1752